src/test/storage/table_test.cpp

```diff
@@ -77,6 +78,36 @@ TEST_F(StorageTableTest, GetColumnIdByName) {
 
 TEST_F(StorageTableTest, GetChunkSize) { EXPECT_EQ(table.target_chunk_size(), 2u); }
 
-TEST_F(StorageTableTest, CompressChunk) { EXPECT_THROW(table.compress_chunk(ChunkID{1}), std::exception); }
+TEST_F(StorageTableTest, CompressChunk) { 
+
+  table.append({4, "Hello"});
+  table.append({6, "Hello"});
+  table.append({3, "!"});
+
+  table.compress_chunk(ChunkID{0});
+
+  auto encoded_chunk = table.get_chunk(ChunkID{0});
+  auto dict_seg_0 = static_cast<DictionarySegment<int>>(encoded_chunk->get_segment(ColumnID{0}));
+  auto dict_seg_1 = static_cast<DictionarySegment<std::string>>(encoded_chunk->get_segment(ColumnID{1}));
```

Comment:
At this point, we recommend testing whether the compressed chunk actually contains segments of type *DictionarySegment* before static casting them and simply assuming that this works.

We found out, that a ``dynamic_cast`` will throw an ``std::bad_cast`` error if you try to cast a *ValueSegment* into a *DictionarySegment*, so using this we were able to check whether the segment type changed with the ``EXPECT_THROW`` and ``EXPECT_NO_THROW`` checks.

Reply:
Makes sense.